ts/session/apis/open_group_api/sogsv3/knownBlindedkeys.ts

```diff
@@ -0,0 +1,413 @@
+  const usUnblinded = UserUtils.getOurPubKeyStrFromCache();
+
+  if (!PubKey.hasBlindedPrefix(blindedOrNakedId)) {
+    return blindedOrNakedId === usUnblinded;
+  }
+  const found = assertLoaded().find(
+    m => m.blindedId === blindedOrNakedId && m.realSessionId === usUnblinded
+  );
+  return Boolean(found);
+}
+
+/**
+ * This function returns the cached blindedId for us, given a public conversation.
+ */
+export function getUsBlindedInThatServer(convo: ConversationModel | string): string | undefined {
+  if (!convo) {
+    return undefined;
+  }
+  const convoId = isString(convo) ? convo : convo.id;
+
+  if (
+    !getConversationController()
+      .get(convoId)
+      ?.isOpenGroupV2()
+  ) {
+    return undefined;
+  }
+  const room = OpenGroupData.getV2OpenGroupRoom(isString(convo) ? convo : convo.id);
+  if (!room || !roomHasBlindEnabled(room) || !room.serverPublicKey) {
+    return undefined;
+  }
+  const usNaked = UserUtils.getOurPubKeyStrFromCache();
+
+  const found = assertLoaded().find(
+    m => m.serverPublicKey === room.serverPublicKey && m.realSessionId === usNaked
+  );
+  return found?.blindedId;
+}
+
+/**
+ * This function can be called to find all blinded conversations we have with a user given its real sessionID.
+ * It should be used when we get a message request response, to merge all convos into one.
+ *
+ * This function is quite resource intensive, so do not call it everywhere
+ */
+function findNotCachedBlindedConvoFromUnblindedKey(
+  unblindedID: string,
+  serverPublicKey: string,
+  sodium: LibSodiumWrappers
+): Array<ConversationModel> {
+  if (PubKey.hasBlindedPrefix(unblindedID)) {
+    throw new Error(
+      'findNotCachedBlindedConvoFromUnblindedKey unblindedID is supposed to be unblinded!'
+    );
+  }
+
+  // we iterate only over the convos private, with a blindedId, and active,
+  // so the one to which we sent a message already or received one from outside sogs.
+  const foundConvosForThisServerPk =
+    getConversationController()
+      .getConversations()
+      .filter(m => m.isPrivate() && PubKey.hasBlindedPrefix(m.id) && m.isActive())
+      .filter(m => {
+        return tryMatchBlindWithStandardKey(unblindedID, m.id, serverPublicKey, sodium);
+      }) || [];
+
+  // we should have only one per server, as we gave the serverpubkey and a blindedId is uniq for a serverPk
+
+  return foundConvosForThisServerPk;
+}
+
+/**
+ * Look for a cached match of that blindedId and that serverPubkey.
+ * This function is expensive and should only be run on some very specific case. You shouldn't need to add any other calls to this function in the app
+ * @param blindedId the blindedId to look for
+ * @param serverPubKey the serverPubkey on which this blindedId is found
+ * @param sodium the sodium instance
+ * @returns the conversationId of the naked private convo found, matching that blindedId on that serverPubkey
+ */
+export async function findCachedBlindedMatchOrItLookup(
```

Comment:
What does `It` mean in this case? Small thing but maybe we could change this to `findCachedBlindedMatchOrDoLookup`

Reply:
this is a typo and was supposed to be `findCachedBlindedMatchOrLookItUp`. I renamed it